zone/bonuses.cpp

```diff
@@ -128,54 +130,87 @@ void Client::CalcBonuses()
 		consume_food_timer.SetTimer(timer);
 }
 
-int Mob::CalcRecommendedLevelBonus(uint8 level, uint8 reclevel, int basestat)
+int Mob::CalcRecommendedLevelBonus(uint8 current_level, uint8 recommended_level, int base_stat)
 {
-	if( (reclevel > 0) && (level < reclevel) )
-	{
-		int32 statmod = (level * 10000 / reclevel) * basestat;
+	if (recommended_level && current_level < recommended_level) {
+		int32 stat_modifier = (current_level * 10000 / recommended_level) * base_stat;
 
-		if( statmod < 0 )
-		{
-			statmod -= 5000;
-			return (statmod/10000);
-		}
-		else
-		{
-			statmod += 5000;
-			return (statmod/10000);
-		}
+		stat_modifier += stat_modifier < 0 ? -5000 : 5000;
+
+		return (stat_modifier / 10000);
 	}
 
 	return 0;
 }
 
-void Client::CalcItemBonuses(StatBonuses* newbon) {
-	//memset assumed to be done by caller.
 
-	// Clear item faction mods
+
+// These item stat caps depend on spells/AAs so we process them after those are processed
+void Client::ProcessItemCaps()
```

Comment:
Strange this PR is showing this as net new when this already exists.

Reply:
Yeah no clue.